eta/core/features.py

```diff
@@ -368,345 +481,526 @@ class CanFeaturizeError(Exception):
-                        # Pre-process and then featurize the frame
-                        _img = self._frame_preprocessor(img)
-                        v = self._frame_featurizer.featurize(_img)
-                    else:
-                        # Featurize the frame
-                        v = self._frame_featurizer.featurize(img)
-
-                    # Write the feature to disk
-                    np.savez_compressed(path, v=v)
-
-                if returnX:
-                    if X is None:
-                        # Lazily build the GrowableArray now that we know the
-                        # dimension of the features
-                        X = GrowableArray(len(v))
-                    X.update(v)
-
-        if self._frame_featurizer and not self._keep_alive:
-            # Stop the frame featurizer
-            self._frame_featurizer.stop()
-            self._frame_featurizer = None
-
-        return X.finalize() if returnX else None
-
-    def featurized_frame_path(self, frame_number):
-        '''Returns the backing path for the given frame number.'''
-        return os.path.join(
-            self._backing_path, self._frame_string % frame_number)
+                if self._frame_preprocessor is not None:
+                    _img = self._frame_preprocessor(img)
+                else:
+                    _img = img
 
-    def flush_backing(self):
-        '''Deletes all existing feautres on disk in the current backing path.
-        The backing directory itself is not deleted.
-        '''
-        files = [
-            f for f in os.listdir(self._backing_path) if f.endswith(".npz")]
-        for f in files:
-            os.remove(os.path.join(self._backing_path, f))
+                v = self._frame_featurizer.featurize(_img)
+                path = self.get_featurized_frame_path(vr.frame_number)
+                self._write_feature(v, path)
 
-    def _stop(self):
-        if self._frame_featurizer:
-            self._frame_featurizer.stop()
-            self._frame_featurizer = None
+    def is_featurized(self, frame_number):
+        '''Returns True/False if the given frame has been featurized.
 
-    def update_backing_path(self, backing_path):
-        '''Update the backing path and create the directory tree, if needed.'''
-        self._backing_path = backing_path
-        try:
-            os.makedirs(self._backing_path)
-        except OSError as e:
-            if e.errno != errno.EEXIST:
-                raise
+        Args:
+            frame_number: the frame number
 
+        Returns:
+            True/False
+        '''
+        path = self.get_featurized_frame_path(frame_number)
+        return os.path.isfile(path)
 
-class ORBFeaturizerConfig(Config):
-    '''Configuration settings for an ORBFeaturizer.'''
+    def get_feature_paths(self):
+        '''Returns a list of absolute paths to the features on disk.'''
+        return etau.list_files(self.backing_dir, abs_paths=True)
```

Comment:
Good point. I added in a basic `.endswith(".npz")` filter.

As a general principle, we should avoid putting more than one type of thing in directories that contain sequences like `%06.<ext>`. For example, `eta.core.utils.parse_dir_pattern` is used heavily throughout the pipeline system, and it will get confused if there are multiple types of sequences in there.